homeassistant/helpers/service.py

```diff
@@ -827,28 +849,34 @@ async def entity_service_call(  # noqa: C901
         for future in done:
             future.result()  # pop exception if have
 
+    # ServiceResult for multiple entities is ambiguous, so just return when
+    # a single entity is involved.
+    return results[0] if len(results) == 1 else None
```

Comment:
An alternative: I'm thinking maybe this should also accept a parameter to opt in to return values, then raise an error if multiple entities are matched earlier in this function.

Reply:
Can we skip updating this method ? Its not needed for script syntax (powering automations/scripts), as that one calls `services.async_call` directly.